libraries/node-core-library/src/RealNodeModulePath.ts

```diff
@@ -103,22 +107,23 @@ export class RealNodeModulePathResolver {
       if (linkTarget) {
         // Relative path in symbolic link. Should be resolved relative to real path of base path.
         const resolvedTarget: string = path.resolve(
-          `${realpathBeforeNodeModules}${input.slice(nodeModulesIndex, linkStart)}`,
+          realpathBeforeNodeModules,
+          input.slice(nodeModulesIndex + 1, linkStart),
           linkTarget
         );
         // Cache the result of the combined resolution to avoid the readlink call in subsequent calls
         cache.set(linkCandidate, resolvedTarget);
         cache.set(resolvedTarget, resolvedTarget);
-        return `${resolvedTarget}${input.slice(linkEnd)}`;
+        return path.join(resolvedTarget, input.slice(linkEnd + 1));
       }
 
       // No symlink, so just return the real path before the last node_modules combined with the
       // subsequent path segments
-      return `${realpathBeforeNodeModules}${input.slice(nodeModulesIndex)}`;
+      return path.join(realpathBeforeNodeModules, input.slice(nodeModulesIndex + 1));
     }
 
     this.realNodeModulePath = (input: string) => {
-      return realNodeModulePathInternal(path.normalize(input));
+      return realNodeModulePathInternal(this._normalizeTrailingSlash(path.resolve(input)));
```

Comment:
If `input` is relative it'll now be resolved relative to the CWD. Expected?

Reply:
By design. `fs.realpathSync.native` when passed a relative path resolves it relative to CWD, so this is consistent with the API it is filling in for.